### ISSUE TITLE
Instantiating racket/gui changes WXME error message

### ISSUE THREAD

**Issue by shhyou** [racket/gui#321]
Is it expected that instantiating `racket/gui/base` changes the error message of reading WXME files?

I have a (problematic) code like below:

```racket
#lang racket/base

(require racket/file racket/port syntax/modread)
; (require racket/gui/base)

(with-module-reading-parameterization
  (lambda ()
    (with-input-from-string (file->string "some_wxme_file.rkt") ;; buggy code
      (lambda () (read-syntax "some_wxme_file.rkt")))))
```

Without `racket/gui/base`, the error message is
> read-syntax: WXME format problem while reading for image-snip image content (expected expected a #\newline when reading bytes, got #\�) from port: #<input-port:string> around position: 32171

But if I uncomment the `(require racket/gui/base)` line, the error message becomes
> insert-file in text%: error loading the file (read-from-file-failed)

Is this the expected behavior?

**Comment by rfindler** [comment]
I can't answer your final question directly but it is the case that there are two implementations of the WXME format; one lives in racket/gui and is wired into `text%` via its saving and loading functionality, and the other one is in [the `wxme` library](https://docs.racket-lang.org/gui/WXME_Decoding.html). The `text%`-based one is the full featured one but it depends on `racket/gui/base` which (as I'm sure you know!) is not available in some contexts (notably linux without X).

**Comment by LiberalArtist** [comment]
It strikes me that the error message without `racket/gui/base` has much more useful information.

**Comment by rfindler** [comment]
Yes, you're right: properly reading the wxme format entails creating GUI objects in the general case (arbitrary snips and, in practice, an interesting one is an editor snip that contains a text object). This mechanism is open, as people can implement new snips and those snips will know how to save themselves. So the "real" implementation is the one that is inside the text% object and it can defer to these later-added snip implementations. There is a need, however, to load wxme files in certain GUI-less contexts, when they are known to contain some limited form of snips where there are substitutes available. That's the role of the wxme library. It works for only some known set of snips and it also supports contexts where one cannot instantiate racket/gui/base.
